starlette/testclient.py

```diff
@@ -77,6 +77,27 @@ def __init__(self, session: WebSocketTestSession) -> None:
         self.session = session
 
 
+class WebSocketReject(WebSocketDisconnect):
+    """
+    A special case of WebSocketDisconnect, raised in the TestClient if the
+    socket is closed before being accepted, either with a send_denial_response()
+    or a websocket.close()
+    """
+
+    def __init__(
+        self,
+        response_status: int,
+        response_headers: typing.List[typing.Tuple[bytes, bytes]] = [],
+        response_body: bytes = b"",
+        close_code: int = 1000,
+        close_reason: typing.Optional[str] = None,
```

Comment:
I think this class should only have non-WebSocket related data i.e. `status_code`, `headers`, and `body`.

Reply:
I'll remove it from the PR.  The intention is for the TestClient to be able to verify the behaviour of the application which is being tested, the arguments that it passes to the close() command.

Reply:
I'll remove the parameters.  But it is still a subclass of WebSocketDisconnect.

Reply:
Right, now I get it.  I had the parameters because I used this class for both kinds of "rejection" response.  The one
done with close(), and the wone with http.response.
Now I raise a WebSocketDisconnect for the close() type reject, but that means that a TestClient won't know the WebSocketDisconnect
is because of a reject.  it looks just like a regular close().  I can live with that.